Implement combineWithAllErrors

tobloef opened the pull request:
This PR implements a utility function `combineWithAllErrors`, which I needed for a project. It basically does the same as `combine`, but instead of returning `Result<[T1, T2, ...], E1 | E2 | ...>` it returns `Result<[T1, T2, ...], (E1 | E2 | ...)[]>`. That is, instead of short-circuiting like the normal `combine`, it returns a list of all the errors it encountered, if the result is an error. It's only for the synchronous `Result` right now, not `ResultAsync`.

This may or may not be something you want to include in the library, but I found it to be very useful, and I believe it might solve #197 as well. We can change the name (or any other aspects) if you wish, I just thought that I would share the changes I made.

supermacro commented:
Hey, @tobloef this is a great idea. 

A few things:

- I just switched over from TravisCI to CircleCI, so I need you to please rebase your changes to the latest commit in this project. 
- If I'm not mistaken / making any wrong assumptions, getting this to work with `ResultAsync` is trivial and only requires a `Promise.all` and `fromSafePromise` call
 - see `combine` for `ResultAsync` (https://github.com/supermacro/neverthrow/blob/master/src/utils.ts#L57)  - Could you please update the docs (specifically the [Utilities Section](https://github.com/supermacro/neverthrow#utilities)) to include this new function. A short example would be great too.

tobloef commented:
Hey @supermacro, I've updated the README, implemented the async version and fixed the stuff the linter was complaining about. Let me know what you think.

supermacro reviewed (approved):
Looks good to me! Thank you for the contribution 🙂
